src/openads/application/service/AddPositionUseCase.js

```diff
@@ -28,12 +32,31 @@ export default class AddPositionUseCase {
     return this._positionRepository.find({id})
       .then(this._filterPositionAlreadyExists)
       .then(() => this._positionFactory.create({id, name, source, placement, segmentation, sizes, native}))
-      .then(position => this._positionRepository.saveOrUpdate({position}))
+      .then(createdPosition => this._setAdToPosition(createdPosition))
+      .then(positionWithAd => this._positionRepository.saveOrUpdate({position: positionWithAd}))
+      .then(savedPosition => this._filterPositionAdIsAvailable(savedPosition))
```

Comment:
we can make filterPositionAdIsAvailable as a function reference like filterPositionAlreadyExists

Reply:
True! done :)